dspace-api/src/main/java/org/dspace/content/ItemServiceImpl.java

```diff
@@ -1324,53 +1338,56 @@ public boolean canCreateNewVersion(Context context, Item item) throws SQLExcepti
                 .getMap().get(relationshipType.getRightLabel());
             otherItem = relationship.getLeftItem();
             relationName = relationship.getRelationshipType().getRightLabel();
         }
 
-        if (hashMaps != null) {
-            resultingMetadataValueList.addAll(handleRelationshipTypeMetadataMappping(item, hashMaps,
+        if (hashMaps != null && enableVirtualMetadata) {
+            resultingMetadataValueList.addAll(handleRelationshipTypeMetadataMappping(context, item, hashMaps,
                                                                                      otherItem, relationName));
         }
         resultingMetadataValueList.add(getRelationMetadataFromOtherItem(otherItem, relationName));
         return resultingMetadataValueList;
     }
 
-    private List<MetadataValue> handleRelationshipTypeMetadataMappping(Item item,
-                                                                       HashMap<String, List<String>> hashMaps,
+    private List<MetadataValue> handleRelationshipTypeMetadataMappping(Context context, Item item,
+                                                                       HashMap<String, VirtualBean> hashMaps,
                                                                        Item otherItem,
-                                                                       String relationName) {
+                                                                       String relationName) throws SQLException {
         List<MetadataValue> resultingMetadataValueList = new LinkedList<>();
-        for (Map.Entry<String, List<String>> entry : hashMaps.entrySet()) {
+        for (Map.Entry<String, VirtualBean> entry : hashMaps.entrySet()) {
             String key = entry.getKey();
-            List<String> value = entry.getValue();
+            VirtualBean virtualBean = entry.getValue();
 
-            MetadataValue metadataValue = constructMetadataValue(key);
-            metadataValue = constructResultingMetadataValue(item, otherItem, value, metadataValue);
-            if (StringUtils.isNotBlank(metadataValue.getValue())) {
-                resultingMetadataValueList.add(metadataValue);
+            for (String value : virtualBean.getValues(context, otherItem)) {
+                MetadataValue metadataValue = constructMetadataValue(key);
+                metadataValue = constructResultingMetadataValue(item, value, metadataValue);
+                if (StringUtils.isNotBlank(metadataValue.getValue())) {
+                    resultingMetadataValueList.add(metadataValue);
+                }
             }
         }
         return resultingMetadataValueList;
     }
 
     private MetadataValue getRelationMetadataFromOtherItem(Item otherItem, String relationName) {
-        MetadataValue metadataValue = constructMetadataValue("relation_" + relationName);
+        MetadataValue metadataValue = constructMetadataValue("relation." + relationName);
```

Comment:
Should we be using the `Enum` here, instead of hardcoding the string "relation"?

Reply:
I've migrated it to the Enum